Property 'getCaption' in type 'Section' is not assignable to the same property in base type 'AbstractBlock'

When using the types defined in `index.d.ts` and running `tsc` with TS 3.7.5 and `strict` tsconfig, we run into:

```
> $(npm bin)/tsc -v
Version 3.7.5

> $(npm bin)/tsc
node_modules/asciidoctor/types/index.d.ts:2190:5 - error TS2416: Property 'getCaption' in type 'Section' is not assignable to the same property in base type 'AbstractBlock'.
  Type '() => string | undefined' is not assignable to type '() => string'.
    Type 'string | undefined' is not assignable to type 'string'.
      Type 'undefined' is not assignable to type 'string'.

2190     getCaption(): string | undefined;
         ~~~~~~~~~~
```

Note that this only happens with a `strict` tsconfig:

```
{
  "compilerOptions": {
    "target": "es2015",
    "module": "commonjs",
    "outDir": "./dist" /,
    "strict": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "typeRoots": ["node_modules/@types", "types"],
    "esModuleInterop": true
  }
}
```

--- a/packages/core/src/asciidoctor-core-api.js
+++ b/packages/core/src/asciidoctor-core-api.js
@@ -394,7 +394,8 @@ AbstractBlock.prototype.getSourceLocation = function () {
  * @memberof AbstractBlock
  */
 AbstractBlock.prototype.getCaption = function () {
-  return this.$caption()
+  var caption = this.$caption()
+  return caption === Opal.nil ? undefined : caption
 }
 
 /**

--- a/packages/core/types/index.d.ts
+++ b/packages/core/types/index.d.ts
@@ -1989,7 +1989,7 @@ export namespace Asciidoctor {
      * Get the caption for this block.
      * @returns the caption for this block
      */
-    getCaption(): string;
+    getCaption(): string | undefined;
 
     /**
      * Set the caption for this block.

--- a/packages/core/types/tests.ts
+++ b/packages/core/types/tests.ts
@@ -537,7 +537,7 @@ class BlogConverter {
     };
   }
 
-  convert(node: Asciidoctor.AbstractNode, transform: string | null, opts: any) {
+  convert(node: Asciidoctor.AbstractNode, transform: string | undefined, opts: any) {
     const template = this.transforms[transform || node.getNodeName()];
     if (template) {
       return template(node);
@@ -578,7 +578,7 @@ assert(callouts.getListIndex() === 2);
 assert(callouts.getCalloutIds(1) === '');
 assert(callouts.getCurrentList().length === 0);
 
-let parsedAttrs: Asciidoctor.Attributes;
+let parsedAttrs: Asciidoctor.Attributes = {};
 const registryAttrs = processor.Extensions.create();
 registryAttrs.block(function() {
   this.named('attrs');
@@ -684,7 +684,7 @@ try {
   const data: any[] = [];
   console.log = function() {
     data.push({method: 'log', arguments});
-    return defaultLog.apply(console, arguments);
+    defaultLog.apply(console, arguments as any);
   };
   const timings = processor.Timings.create();
   processor.convert('Hello *world*', {timings});

--- a/packages/core/types/tsconfig.json
+++ b/packages/core/types/tsconfig.json
@@ -8,8 +8,7 @@
     "esModuleInterop": true,
     "noImplicitAny": true,
     "noImplicitThis": true,
-    "strictNullChecks": false,
-    "strictFunctionTypes": true,
+    "strict": true,
     "baseUrl": "../",
     "paths": {
       "@asciidoctor/core": [

